scispacy/custom_sentence_segmenter.py

```diff
@@ -31,32 +15,23 @@ def combined_rule_sentence_segmenter(doc: Doc) -> Doc:
 
     @param doc: the spaCy document to be annotated with sentence boundaries
     """
-    segmenter = pysbd.Segmenter(language='en', clean=False)
-    segments = segmenter.segment(doc.text)
-    segments = merge_segments(segments)
-
-    # pysbd splits raw text into sentences, so we have to do our best to align those
-    # segments with spacy tokens
-    segment_index = 0
-    current_segment = segments[segment_index]
-    built_up_sentence = ""
-    for i, token in enumerate(doc):
-        if i == 0 and (token.is_space or token.text == '.'):
-            token.is_sent_start = True
-            continue
-        if token.text.replace('\n', '').replace('\r', '') == '':
-            token.is_sent_start = False
-        elif len(built_up_sentence) >= len(current_segment):
+    segmenter = pysbd.Segmenter(language="en", clean=False, char_span=True)
+    sents_char_spans = segmenter.segment(doc.text)
+
+    char_spans = [
+        doc.char_span(sent_span.start, sent_span.end)
+        for sent_span in sents_char_spans
+    ]
+    start_token_ids = [span[0].idx for span in char_spans if span is not None]
+    for token in doc:
+        prev_token = token.nbor(-1) if token.i != 0 else None
+        if token.idx in start_token_ids:
+            if prev_token and prev_token.text in ABBREVIATIONS:
+                token.is_sent_start = False
+            else:
+                token.is_sent_start = True
+        elif prev_token and prev_token.i != 0 and prev_token.text.count('\n') >= 2:
```

Comment:
Will want to adjust the docstring to reflect what you are doing with new lines here.

Reply:
Again, had to adjust sentencing as per `custom_tokenizer` which handles newlines differently.

Using only pysbd 
```
import pysbd
text = "\n\n2 Long Short-Term Memory Networks\n\n\n\n2.1 Overview\n\nRecurrent neural networks (RNNs) are able to process input sequences of arbitrary leng
th via the recursive application of a transition function on a hidden state vector ht."
seg = pysbd.Segmenter(language="en", clean=False)
```
> ['2 Long Short-Term Memory Networks',
 '2.1 Overview',
 'Recurrent neural networks (RNNs) are able to process input sequences of arbitrary length via the recursive application of a transition function on a hidden state vector ht.']

Reply:
& this line does what is already written in docstring.
> New lines that separate sentences will be added to the beginning of the next sentence.

Reply:
:+1: